push-down-test/run-tests.sh

```diff
@@ -0,0 +1,229 @@
+  pd_no_batch_pid=$!
+
+  "$pd_bin" -config "$push_down_with_batch_config_dir"/pd.toml -log-file "$pd_with_batch_log_file" -L ${log_level} &
+  pd_with_batch_pid=$!
+}
+
+function run_tikvs() {
+  "$tikv_bin" -C "$push_down_no_batch_config_dir"/tikv.toml --log-file "$tikv_no_batch_log_file" -L ${log_level} &
+  tikv_no_batch_pid=$!
+
+  "$tikv_bin" -C "$push_down_with_batch_config_dir"/tikv.toml --log-file "$tikv_with_batch_log_file" -L ${log_level} &
+  tikv_with_batch_pid=$!
+}
+
+function run_tidbs() {
+  export GO_FAILPOINTS="github.com/pingcap/tidb/expression/PushDownTestSwitcher=return(\"\")"
+  "$tidb_bin" -log-file "$no_push_down_tidb_log_file" -config "$no_push_down_config_dir"/tidb.toml -L ${log_level} &
+  tidb_no_push_down_pid=$!
+
+  export GO_FAILPOINTS="github.com/pingcap/tidb/expression/PushDownTestSwitcher=return(\"all\")"
+  "$tidb_bin" -log-file "$push_down_no_batch_tidb_log_file" -config "$push_down_no_batch_config_dir"/tidb.toml -L ${log_level} &
+  tidb_push_down_no_batch_pid=$!
+
+  export GO_FAILPOINTS="github.com/pingcap/tidb/expression/PushDownTestSwitcher=return(\"all\")"
+  "$tidb_bin" -log-file "$push_down_with_batch_tidb_log_file" -config "$push_down_with_batch_config_dir"/tidb.toml -L ${log_level} &
+  tidb_push_down_with_batch_pid=$!
+}
+
+# make sure that after run this function,
+# the current working dir is not change
+function build_tidb() {
+  if [ -z $tidb_src_dir ]; then
+    echo "TiDB source code path not specified (tidb_src_dir environment variable not set)"
+    readonly tidb_src_url="https://github.com/pingcap/tidb/archive/master.zip"
+    echo "Retrieve TiDB source code from ${tidb_src_url}"
+    mkdir -p /tmp/copr_test/
+    rm -rf /tmp/copr_test/tidb_master /tmp/copr_test/tidb_master.zip
+    wget ${tidb_src_url} -O /tmp/copr_test/tidb_master.zip
+    unzip /tmp/copr_test/tidb_master.zip -d /tmp/copr_test/tidb_master
+    tidb_src_dir=/tmp/copr_test/tidb_master/tidb-master
+  fi
+
+  cur_dir=$(pwd)
+  cd ${tidb_src_dir}
+  make failpoint-enable
+  make
+  tidb_bin=$(pwd)/bin/tidb-server
+  cd ${cur_dir}
+}
+
+function build_tester() {
+  go build -o "$push_down_test_bin" ./src
+}
+
+function cat_log() {
+  echo "no push down tidb log (this tidb use mocktikv, so has no pd and tikv log)"
+  cat "$no_push_down_tidb_log_file"
+  echo
+  echo
+
+  echo "push down no batch tidb log"
+  cat "$push_down_no_batch_tidb_log_file"
+  echo
+  echo "push down no batch tikv log"
+  cat "$tikv_no_batch_log_file"
+  echo
+  echo "push down no batch pd log"
+  cat "$pd_no_batch_log_file"
+  echo
+  echo
+
+  echo "push down with batch tidb log"
+  cat "$push_down_with_batch_tidb_log_file"
+  echo
+  echo "push down with batch tikv log"
+  cat "$tikv_with_batch_log_file"
+  echo
+  echo "push down with batch pd log"
+  cat "$pd_with_batch_log_file"
+  echo
+  echo
+}
+
+function kill_all_proc() {
+  killall -TERM tidb-server
+  killall -TERM tikv-server
+  killall -TERM pd-server
+}
```

Comment:
Using ` pgrep tidb\* | xargs kill -TERM` is better. Because in localhost, the use may has multiple tidb-bin to test, and their name are not all `tidb-server`. So does tikv-server and pd-server

Reply:
However, pd\* will match many processor that is not pd-server. So maybe we should use killall -TERM pd-server here.